### PR TITLE
Removed repetitive parsing of JSON during backup

### PR DESCRIPTION
**Pull request by sdhiman99**
#95 Fixed creation of backup
Please take a look and tell if this approach seems alright. This is to convert JSON string with escape sequences to regular JSON string ready to be parsed.
`"{\"key\":\"value\"}"`  -> `{"key":"value"}`

### CODE REVIEW

**Comment by b-m-f** [comment]
Hi @sdhiman99 ,

I think you should better take a look at the backup creation.

The 2 important questions:
- is it using `JSON.stringify`
- how is the state stored in local storage before being backed up

Then, instead of replacing everything manually you could remove one of the `JSON.stringify` :)

File: Interface/src/Widgets/Dashboard/Settings.vue [hunk 1]
```diff
@@ -53,9 +53,7 @@ export default {
       element.setAttribute(
         "href",
         "data:application/json;charset=utf-8," +
-          encodeURIComponent(
-            JSON.stringify(window.localStorage.getItem("vuex"))
-          )
+          encodeURIComponent(window.localStorage.getItem("vuex"))
```
Comment (b-m-f) [inline Interface/src/Widgets/Dashboard/Settings.vue:56]: Ah nice!

**Comment by b-m-f** [comment]
Looks good.

Would you be able to add tests for this?

It should look something like this:

- add server
- create backup
- verify that the downloaded file contains 1 time stringified content

You can take [this test](https://github.com/b-m-f/WirtBot/blob/6938c6c5f18a056069cc2359bbc53ad6c90b2493/System-Tests/tests/nightwatch/dashboard/dashboard_device_widget.spec.js#L70) as an inspiration.

Having the test will help ion keeping things under control in the future

**Comment by sdhiman99** [comment]
Sure! I'll commit the tests soon.

**Comment by sdhiman99** [comment]
Hi @b-m-f,
Please take a look and suggest changes. Here, I've used a `glob` because it is difficult to get the filename beforehand. So, it searches for `dasboard-backup-*.json` pattern and gets the latest file matching it. Next step is to parse the `json` once and check if a key in it, lets take `version`, is a number or not.

File: System-Tests/tests/nightwatch/dashboard/dashboard_device_widget.spec.js [hunk 2]
```diff
@@ -83,6 +84,27 @@ module.exports = {
             throw new Error(error);
         }
     },
+    "Download and verify backup": async function (browser) {
```
Comment (b-m-f) [inline System-Tests/tests/nightwatch/dashboard/dashboard_device_widget.spec.js:87]: `"Download and verify backup"` -> "Download and verify that backup is a simple JSON string"

File: System-Tests/tests/nightwatch/dashboard/dashboard_device_widget.spec.js [hunk 3]
```diff
@@ -83,6 +84,27 @@ module.exports = {
...
+        await browser.click("#export button");
+        try {
+            await setTimeoutAsync(1000, 'waitForDownload').then(async () => {
+                let files = glob.sync("/tmp/WirtTestDownloads/dasboard-backup-*.json");
+                let file = files[files.length - 1];
+                // console.log(file);
```
Comment (b-m-f) [inline System-Tests/tests/nightwatch/dashboard/dashboard_device_widget.spec.js:93]: Can this be removed?

**Comment by b-m-f** [comment]
@sdhiman99 

looks good. Just a few small things left.

& `glob` seems like a helpful tool here. Lets just pin the version.

**Review by b-m-f — APPROVED** [review]

**Comment by sdhiman99** [comment]
Hi @b-m-f,
I noticed that you removed the `hacktoberfest` tag from the repo. That actually made this PRs ineligible for participation. Since, the season is now over and I cannot participate, can you add the tag for couple more days or maybe tag this particular PR with `hacktoberfest-accepted`?

**Comment by b-m-f** [comment]
@sdhiman99 of course! Sorry about that and thanks again for the help.

Enjoy your shirt or tree! :+1:

**Comment by sdhiman99** [comment]
Thanks @b-m-f. But, this PR is still ineligible. I think they left out this boundary case, as repos are no longer being monitored for `hacktoberfest` tag. Anyways, I've got some spare ones that may help. 👍

**Pull request merged**